agent/engine/dependencygraph/graph.go

```diff
@@ -175,16 +191,23 @@ func verifyStatusResolvable(target *api.Container, existingContainers map[string
 	return true
 }
 
-func verifyTransitionDependenciesResolved(target *api.Container, existingContainers map[string]*api.Container) bool {
+func verifyTransitionDependenciesResolved(target *api.Container,
+	existingContainers map[string]*api.Container,
+	existingResources map[string]taskresource.TaskResource) error {
 	targetGoal := target.GetDesiredStatus()
 	if targetGoal >= api.ContainerStopped {
 		// A container can always stop, die, or reach whatever other state it
 		// wants regardless of what dependencies it has
-		return true
+		return nil
 	}
 
-	// TODO: add verifyResourceDependenciesResolved check here
-	return verifyContainerDependenciesResolved(target, existingContainers)
+	if !verifyContainerDependenciesResolved(target, existingContainers) {
+		return ErrContainerDependencyNotResolved
+	}
+	if !verifyResourceDependenciesResolved(target, existingResources) {
+		return ErrResourceDependencyNotResolved
```

Comment:
Can you point me to the unit test where this path is being covered? I'm mostly looking for the place where the test verifies that this error (`ErrResourceDependencyNotResolved`) is returned when resource dependency is not satisfied.

Reply:
There are tests for the method `verifyResourceDependenciesResolved` itself which returns a bool. I can move them up a method to check for the returned error.

Reply:
I think there's value in having a test that verifies the error returned as well. Feel free to either modify the existing one or to add a new one scoped just for that.

Reply:
updated.